command.go

```diff
@@ -328,12 +328,17 @@ func stripFlags(args []string, c *Command) []string {
 	return commands
 }
 
-func argsMinusX(args []string, x string) []string {
+// argsMinusFirstX removes only the first x from args.  Otherwise, commands that look like
+// openshift admin policy add-role-to-user admin my-user, lose the admin argument (arg[4]).
+func argsMinusFirstX(args []string, x string) []string {
+	removedFirstX := false
 	newargs := []string{}
 
 	for _, y := range args {
-		if x != y {
+		if removedFirstX || (x != y) {
 			newargs = append(newargs, y)
+		} else {
+			removedFirstX = true
```

Comment:
@deads2k yeah, so I guess my version was actually mutating the original slice, rather than creating a new slice with just the correct data.  Thanks for the new (working) version!